src/HDInsight/HDInsight/ManagementCommands/NewAzureHDInsightClusterCommand.cs

```diff
@@ -664,20 +665,20 @@ private Guid GetApplicationId(Guid applicationId)
                 return applicationId;
             }
 
-            GraphRbacManagementClient graphClient = AzureSession.Instance.ClientFactory.CreateArmClient<GraphRbacManagementClient>(
+            MicrosoftGraphClient graphClient = AzureSession.Instance.ClientFactory.CreateArmClient<MicrosoftGraphClient>(
                 DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.Graph);
 
             graphClient.TenantID = DefaultProfile.DefaultContext.Tenant.Id.ToString();
 
-            Microsoft.Azure.Graph.RBAC.Version1_6.Models.ServicePrincipal sp = null;
+            MicrosoftGraphServicePrincipal sp = null;
             try
             {
-                sp = graphClient.ServicePrincipals.Get(ObjectId.ToString());
+                sp = graphClient.ServicePrincipals.GetServicePrincipalWithHttpMessagesAsync(ObjectId.ToString()).Result.Body;//graphClient.ServicePrincipals.Get(ObjectId.ToString());
```

Comment:
You can use this extension method to simplify code: https://github.com/Azure/azure-powershell-common/blob/main/src/Graph.Rbac/MicrosoftGraph/Version1_0/Applications/ServicePrincipalsOperationsExtensions.cs#L210
```suggestion
                sp = graphClient.ServicePrincipals.GetServicePrincipal(ObjectId.ToString());
```

Reply:
Changed